core/unit/name.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package unit
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/juju/juju/internal/errors"
+)
+
+const (
+	// applicationSnippet is a non-compiled regexp that can be composed with
+	// other snippets to form a valid application regexp.
+	applicationSnippet = "(?:[a-z][a-z0-9]*(?:-[a-z0-9]*[a-z][a-z0-9]*)*)"
```

Comment:
**nitpick:** Maybe it is me but snippet doesn't match in my internal thesaurus. snippet means abstract for me.

Here it is a Pattern

Reply:
Yes, pattern is clearer imo